src/js/Color.js

```diff
@@ -0,0 +1,537 @@
+            suggestedColors.contrast = /** @type {!string} */ (newContrastRatio.toFixed(2));
+            colors[desiredLabel] = /** @type {axs.color.SuggestedColors} */ (suggestedColors);
+            continue;
+        }
+
+        var desiredBgLuminance = axs.color.luminanceFromContrastRatio(fgLuminance, desiredContrast + 0.02, !fgLuminanceIsHigher);
+        if (desiredBgLuminance <= 1 && desiredBgLuminance >= 0) {
+            var newBgColor = axs.color.translateColor(bgYCC, desiredBgLuminance);
+            var newContrastRatio = axs.color.calculateContrastRatio(fgColor, newBgColor);
+            var suggestedColors = {};
+            suggestedColors.bg = /** @type {!string} */ (axs.color.colorToString(newBgColor));
+            suggestedColors.fg = /** @type {!string} */ (axs.color.colorToString(fgColor));
+            suggestedColors.contrast = /** @type {!string} */ (newContrastRatio.toFixed(2));
+            colors[desiredLabel] = /** @type {axs.color.SuggestedColors} */ (suggestedColors);
+        }
+    }
+    return colors;
+};
+
+/**
+ * Combine the two given color according to alpha blending.
+ * @param {axs.color.Color} fgColor
+ * @param {axs.color.Color} bgColor
+ * @return {axs.color.Color}
+ */
+axs.color.flattenColors = function(fgColor, bgColor) {
+    var alpha = fgColor.alpha;
+    var r = ((1 - alpha) * bgColor.red) + (alpha * fgColor.red);
+    var g = ((1 - alpha) * bgColor.green) + (alpha * fgColor.green);
+    var b = ((1 - alpha) * bgColor.blue) + (alpha * fgColor.blue);
+    var a = fgColor.alpha + (bgColor.alpha * (1 - fgColor.alpha));
+
+    return new axs.color.Color(r, g, b, a);
+};
+
+/**
+ * Multiply the given vector by the given matrix.
+ * @param {Array.<Array.<number>>} matrix A 3x3 matrix
+ * @param {Array.<number>} vector A 3-element vector
+ * @return {Array.<number>} A 3-element vector
+ */
+axs.color.multiplyMatrixVector = function(matrix, vector) {
+    var a = matrix[0][0];
+    var b = matrix[0][1];
+    var c = matrix[0][2];
+    var d = matrix[1][0];
+    var e = matrix[1][1];
+    var f = matrix[1][2];
+    var g = matrix[2][0];
+    var h = matrix[2][1];
+    var k = matrix[2][2];
+
+    var x = vector[0];
+    var y = vector[1];
+    var z = vector[2];
+
+    return [
+        a*x + b*y + c*z,
+        d*x + e*y + f*z,
+        g*x + h*y + k*z
+    ];
+};
+
+/**
+ * Convert a given RGB color to YCC.
+ * @param {axs.color.Color} color
+ * @return {axs.color.YCC}
+ */
+axs.color.toYCC = function(color) {
+    var rSRGB = color.red / 255;
+    var gSRGB = color.green / 255;
+    var bSRGB = color.blue / 255;
+
+    var r = rSRGB <= 0.03928 ? rSRGB / 12.92 : Math.pow(((rSRGB + 0.055)/1.055), 2.4);
+    var g = gSRGB <= 0.03928 ? gSRGB / 12.92 : Math.pow(((gSRGB + 0.055)/1.055), 2.4);
+    var b = bSRGB <= 0.03928 ? bSRGB / 12.92 : Math.pow(((bSRGB + 0.055)/1.055), 2.4);
+
+    return new axs.color.YCC(axs.color.multiplyMatrixVector(axs.color.YCC_MATRIX, [r, g, b]));
```

Comment:
It looks like we have a loose convention in the surrounding methods to assign a result variable and return that. Not sure if we care much, but it might be more consistent. What do you think?